pkg/operation/botanist/health_check.go

```diff
@@ -816,6 +813,31 @@ func (b *Botanist) healthChecks(initializeShootClients func() error, thresholdMa
 	return apiserverAvailability, controlPlane, nodes, systemComponents
 }
 
+/*This function is created because when Loki is deployed for the first time on already existing shoot
+the maintenance window will not allow Loki StatfulSet to be deployed and the health check will fail.
+Do we are preventing this health check up to a certain version when Loki will be still integrated to the
+landscapes. In versions above 1.8.0 this function can be removed */
+func (b *Botanist) isLoggingHealthCheckNeeded() bool {
+	if b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting {
+		return false
+	}
+	gardenerVersion18, err := semver.NewVersion("1.8.0")
+	if err != nil {
+		return false
+	}
+
+	lastGardenerVersion, err := semver.NewVersion(b.Shoot.Info.Status.Gardener.Version)
+	if err != nil {
+		return false
+	}
+
+	if lastGardenerVersion.Compare(gardenerVersion18) >= 0 {
```

Comment:
You could use something like to match any 1.8.x version.
```golang
	c, err := semver.NewConstraint(">=1.8")
	if err != nil {
		return false
	}

	lastGardenerVersion, err := semver.NewVersion(b.Shoot.Info.Status.Gardener.Version)
	if err != nil {
		return false
	}
	return c.Check(lastGardenerVersion)
```

Reply:
ok